src/main/java/com/kramekk1/medicalClinic/MedicalClinicExceptionHandler.java

```diff
@@ -0,0 +1,18 @@
+package com.kramekk1.medicalClinic;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.time.LocalDateTime;
+
+@ControllerAdvice
+public class MedicalClinicExceptionHandler {
+
+    @ExceptionHandler(MedicalClinicException.class)
+    protected ResponseEntity<ErrorMessage> handlePatientNotFoundException(PatientNotFoundException ex) {
```

Comment:
Zauważ, że piszesz, że jest to @ExceptionHandler(MedicalClinicException.class) dla typu MedicalClinicException natomiast w argumencie dostajesz PatientNotFoundException -> jest to niepoprawne poniewaz nie każdy MedicalClinicException to PatientNotFoundException. W argumencie metody powinieneś przyjąć PatientNotFoundException do tego do zmiany nazwa metody

Reply:
Możesz wygenerować taki konstruktor prywatny bezargumentowy z uzyciem lomboka nad klasą:
@NoArgsConstructor(access = AccessLevel.PRIVATE)